freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UmsUsersStateProvider.java

```diff
@@ -129,95 +133,38 @@ private WorkloadCredential getCredentials(String userCrn, Optional<String> reque
+            grpcUmsClient.listGroupsForMember(IAM_INTERNAL_ACTOR_CRN, accountId, memberCrn, requestId).forEach(gcrn -> {
+                FmsGroup group = crnToFmsGroup.get(gcrn);
+                // If the group is null, then there has been a group membership change after we started the sync
+                // the group and group membership will be updated on the next sync
+                if (group != null) {
+                    usersStateBuilder.addMemberToGroup(group.getName(), username);
+                }
+            });
 
-                List<String> workloadAdministrationGroupNames = rightsResponse.getWorkloadAdministrationGroupNameList();
-                LOGGER.debug("workloadAdministrationGroupNameList = {}", workloadAdministrationGroupNames);
-                workloadAdministrationGroupNames.forEach(groupName -> {
-                    usersStateBuilder.addGroup(nameToGroup(groupName));
-                    usersStateBuilder.addMemberToGroup(groupName, fmsUser.getName());
-                });
+            // Since this user is eligible, add this user to internal group
+            usersStateBuilder.addMemberToGroup(UserServiceConstants.CDP_USERSYNC_INTERNAL_GROUP, username);
 
-                if (isEnvironmentAdmin(environmentCrn, rightsResponse)) {
-                    // TODO: introduce a flag for adding admin
-                    usersStateBuilder.addMemberToGroup("admins", fmsUser.getName());
-                }
+            UserManagementProto.ListWorkloadAdministrationGroupsForMemberResponse workloadAdministrationGroups =
+                    grpcUmsClient.listWorkloadAdministrationGroupsForMember(IAM_INTERNAL_ACTOR_CRN, memberCrn, requestId);
+            workloadAdministrationGroups.getWorkloadAdministrationGroupNameList().forEach(groupName -> {
+                usersStateBuilder.addGroup(nameToGroup(groupName));
+                usersStateBuilder.addMemberToGroup(groupName, username);
+            });
 
-                // get credentials
-                umsUsersStateBuilder.addWorkloadCredentials(fmsUser.getName(), getCredentials(memberCrn, requestId));
-            }
-        } catch (StatusRuntimeException e) {
-            // NOT_FOUND errors indicate that a user/machineUser has been deleted after we have
-            // retrieved the list of users/machineUsers from the UMS. Treat these users as if
-            // they do not have the right to access this environment and belong to no groups.
-            if (e.getStatus().getCode() == Status.Code.NOT_FOUND) {
-                LOGGER.warn("CRN {} not found in UMS. Treating as if CRN has no rights to environment {}: {}",
-                        memberCrn, environmentCrn, e.getLocalizedMessage());
-            } else {
-                throw e;
+            if (environmentAccessRights.hasAdminFreeIPARight()) {
+                usersStateBuilder.addMemberToGroup(ADMIN_FREEIPA_GROUP, username);
             }
+
+            // get credentials
```

Comment:
I think you can drop this command or move this to a method with a self explaining name

Reply:
removed